torch/_functorch/autograd_function.py

```diff
@@ -240,11 +252,167 @@ def custom_function_call_vmap(interpreter, autograd_function, *operands):
     return output
 
 
+def custom_function_call_vmap_generate_rule(interpreter, autograd_function, *operands):
+    unwrapped_operands, in_dims = unwrap_batched(operands, interpreter.level())
+    vmapped_function, get_out_dims = vmapify_autograd_function(
+        autograd_function, in_dims, interpreter.batch_size(), interpreter.randomness())
+
+    with interpreter.lower():
+        output = custom_function_call(vmapped_function,
+                                      True,  # generate_vmap_rule
+                                      *unwrapped_operands)
+
+    out_dims = get_out_dims()
+    return wrap_batched(output, out_dims, interpreter.level())
+
+
 @custom_function_call.py_impl(TransformType.Functionalize)
-def custom_function_call_functionalize(interpreter, autograd_function, *operands):
+def custom_function_call_functionalize(interpreter, autograd_function, generate_vmap_rule, *operands):
```

Comment:
Instead of having to pass this, could we just check its value on the autograd_function?

Reply:
Yeah, that's a better idea, thanks for the suggestion

Reply:
Updated